vignettes/BiocParallel_BatchtoolsParam.Rmd

```diff
@@ -0,0 +1,229 @@
+    job scheduler. It defines how the jobs are submitted to the job
+    scheduler. If the template is not given and the cluster is chosen, a
+    default template is selected from the batchtools package.
+
+     + `log` The log option is logical, TRUE/FALSE. If it is set to TRUE, then
+    the logs which are in the registry are copied to directory given by
+    the user using the `logdir` argument.
+
+    + `logdir` Path to the logs. It is given only if `log=TRUE`.
+
+    + `resultdir` Path to the directory is given when the job has files to be saved in
+    a directory.
+
+# Defining templates
+
+The job submission template controls how the job is processed by the job
+scheduler on the cluster. Obviously, the format of the template will
+differ depending on the type of job scheduler. Let's look at the default
+SLURM template as an example:
+
+```{r}
+fname <- batchtoolsTemplate("slurm")
+cat(readLines(fname), sep="\n")
+```
+
+The `<%= =>` blocks are automatically replaced by the values of the elements in
+the `resources` argument in the`BatchtoolsParam` constructor. Failing to specify critical parameters
+properly (e.g., wall time or memory limits too low) will cause jobs to
+crash, usually rather cryptically. We suggest setting parameters
+explicitly to provide robustness to changes to system defaults. Note
+that `<%= =>` the blocks themselves do not usually need to be modified in the
+template.
+
+The part of the template that is most likely to require explicit
+customization is the last line containing the call to `Rscript`. A more
+customized call may be necessary if the R installation is not standard,
+e.g., if multiple versions of R have been installed on a cluster. For
+example, one might use instead:
+
+````{verbatim}
+    echo 'batchtools::doJobCollection("<%= uri %>")' |\
+        ArbitraryRcommand --no-save --no-echo
+````
+
+If such customization is necessary, we suggest making a local copy of
+the `BiocParallelParam` template, modifying it as required, and then constructing a object
+with the modified template using `template` the argument. However, we find that the
+default templates accessible with `batchtoolsTemplate` are satisfactory in most cases.
+
+# Use cases
+
+As an example for a BatchtoolParam job being run on an SGE cluster, we
+use the same `piApprox` function as defined earlier. The example runs the function
+on 5 workers and submits 100 jobs to the SGE cluster.
+
+```{r simple_sge_example, eval=FALSE}
+library(BiocParallel)
+
+## Pi approximation
+piApprox <- function(n) {
+    nums <- matrix(runif(2 * n), ncol = 2)
+    d <- sqrt(nums[, 1]^2 + nums[, 2]^2)
+    4 * mean(d <= 1)
+}
+
+template <- system.file(
+    package = "BiocParallel",
+    "unitTests", "test_script", "test-sge-template.tmpl"
+)
+param <- BatchtoolsParam(workers=5, cluster="sge", template=template)
+
+## Run parallel job
+result <- bplapply(rep(10e5, 100), piApprox, BPPARAM=param)
+```
+
+Example of SGE with minimal code:
```

Comment:
Ahh, I see what happened. I think that the captions got moved down a section. This text will be moved above the previous code chunk. Here, delete this line and replace with: "Example of SGE demonstrating some of `BatchtoolsParam` methods."